### PR TITLE
Add support for links in Monitoring AlertPolicy

### PR DESCRIPTION
**Pull request by z-nand**
Resolves https://buganizer.corp.google.com/issues/318732744

Adds support for links in AlertPolicy.

I acknowledge that I have:

Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
[Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [make test and make lint](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
[Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.
Release Note Template for Downstream PRs (will be copied)

**Release Note Template for Downstream PRs (will be copied)**

```
monitoring: added `links` field to `google_monitoring_alert_policy` resource
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 1]
```diff
@@ -329,6 +329,10 @@ resource "google_monitoring_alert_policy" "full" {
     content   = "test content"
     mime_type = "text/markdown"
     subject = "test subject"
+	links {
+		displayname = "link display name"
+		url = "http://mydomain.com"
+	}
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:335]: Can you adjust the indentations here and other places to align with others? Two white spaces  are used for one indentation.

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 2]
```diff
@@ -360,6 +364,14 @@ resource "google_monitoring_alert_policy" "mql" {
     content   = "test content"
     mime_type = "text/markdown"
     subject = "test subject"
+	links {
+		displayname = "link display name"
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:368]:
```suggestion
		displayName = "link display name"
```

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 3]
```diff
@@ -329,6 +329,10 @@ resource "google_monitoring_alert_policy" "full" {
...
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:333]:
```suggestion
		displayName = "link display name"
```

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 4]
```diff
@@ -360,6 +364,14 @@ resource "google_monitoring_alert_policy" "mql" {
...
+		url = "http://mydomain.com"
+	}
+	links {
+		displayname = "link display name2"
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:372]:
```suggestion
		displayName = "link display name2"
```

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 5]
```diff
@@ -395,7 +407,11 @@ resource "google_monitoring_alert_policy" "log" {
   documentation {
     content   = "test content"
     mime_type = "text/markdown"
-    subject = "test subject"    
+    subject = "test subject"
+	links {
+		displayname = "link display name"
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:412]:
```suggestion
		displayName = "link display name"
```

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 6]
```diff
@@ -457,6 +473,10 @@ resource "google_monitoring_alert_policy" "promql" {
     content   = "test content"
     mime_type = "text/markdown"
     subject = "test subject"
+	links {
+		displayname = "link display name"
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:477]:
```suggestion
		displayName = "link display name"
```

**Review by zli82016 — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go (hunk 3 above)
Comment (z-nand) [reply mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:333]: commit suggestion

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go (hunk 1 above)
Comment (zli82016) [reply mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:335]: It looks like the indentations are not adjusted.

Comment (z-nand) [reply mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go]: Hi Zhenhua,

Sorry I didn't realize resoving a comment will trigger your review. Indents should be adjusted now!

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go (hunk 6 above)
Comment (z-nand) [reply mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go]: > 

You suggested "displayName" but I think "display_name" should be the correct format according to other "display_name" fields. 
Can you confirm? Thank you!

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go (hunk 1 above)
Comment (zli82016) [reply mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go]: no worries. Thanks.

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 7]
```diff
@@ -329,6 +329,10 @@ resource "google_monitoring_alert_policy" "full" {
     content   = "test content"
     mime_type = "text/markdown"
     subject = "test subject"
+	links {
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:332]:
```suggestion
  links {
```

Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:332]: nits: indentation

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 8]
```diff
@@ -329,6 +329,10 @@ resource "google_monitoring_alert_policy" "full" {
...
+        display_name = "link display name"
+        url = "http://mydomain.com"
+	}
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:335]: nits: indentation

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 9]
```diff
@@ -360,6 +364,14 @@ resource "google_monitoring_alert_policy" "mql" {
     content   = "test content"
     mime_type = "text/markdown"
     subject = "test subject"
+	links {
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:367]: nits: indentation

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 10]
```diff
@@ -360,6 +364,14 @@ resource "google_monitoring_alert_policy" "mql" {
...
+        display_name = "link display name"
+        url = "http://mydomain.com"
+	}
+	links {
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:371]: nits: indentation

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 11]
```diff
@@ -360,6 +364,14 @@ resource "google_monitoring_alert_policy" "mql" {
...
+        display_name = "link display name2"
+        url = "http://mydomain2.com"
+	}
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:374]: nits: indentation

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 12]
```diff
@@ -457,6 +469,10 @@ resource "google_monitoring_alert_policy" "promql" {
     content   = "test content"
     mime_type = "text/markdown"
     subject = "test subject"
+	links {
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:472]: nits: indentation

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 13]
```diff
@@ -457,6 +469,10 @@ resource "google_monitoring_alert_policy" "promql" {
...
+        display_name = "link display name"
+        url = "http://mydomain.com"
+	}
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:475]: nits: indentation

**Review by zli82016 — CHANGES_REQUESTED** [review]
Mostly LGTM.

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go (hunk 7 above)
Comment (z-nand) [reply mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go]: My bad! All the indentations should be fixed now.

Sorry for the trouble!

**Review by zli82016 — APPROVED** [review]
LGTM

**Comment by AmandaHassoun** [comment]
hey @zli82016, how is this supposed to work? I have something like this configured in my Terraform alert policy and I don't see my links anywhere in the GCloud UI nor in the alerts when they fire in Slack:

```
documentation {
    content = "TO FILL"

    links {
      display_name = "runbook"
      url          = "https://github.com/.."
    }
    links {
      display_name = "ui"
      url          = "https://some-uril/ui"
    }
    links {
      display_name = "grafana"
      url          = "https://grafana-prod/...."
    }
  }
```

**Comment by zli82016** [comment]
@AmandaHassoun , do you mind opening a new Github issue https://github.com/hashicorp/terraform-provider-google/issues/new/choose? Thanks.

**Pull request merged**